Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnSlide.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class OnSlide : OnMovementSpecial
+{
+
+    [SerializeField]
+    private float wallSlideUpForce;
+    [SerializeField]
+    private float wallSlideDownForce;
+    [SerializeField]
+    private float wallSlideUpDecay;
+    [SerializeField]
+    private float wallSlideDownDecay;
+    [SerializeField]
+    private Vector2 groundSlideForce;
+    [SerializeField]
+    private float groundSlideDecay;
+
+    [SerializeField]
+    private PlayRandomSource wallSlideSound;
+
+    private bool goingRight;
+
+    private bool goingUp;
+
+    private Vector3 appliedForce;
```

Comment:
If you follow my other comments, this should probably be renamed. If not, this shouldn't be a class level variable